pkgs/by-name/av/av1an/package.nix

```diff
@@ -0,0 +1,73 @@
+{
+  lib,
+  symlinkJoin,
+  makeBinaryWrapper,
+  testers,
+  nix-update-script,
+  av1an-unwrapped,
+  av1an,
+  ffmpeg,
+  python3Packages,
+  libaom,
+  svt-av1,
+  rav1e,
+  libvpx,
+  x264,
+  x265,
+  libvmaf,
+  withAom ? true, # AV1 reference encoder
+  withSvtav1 ? false, # AV1 encoder/decoder (focused on speed and correctness)
+  withRav1e ? false, # AV1 encoder (focused on speed and safety)
+  withVpx ? true, # VP8 & VP9 de/encoding
+  withX264 ? true, # H.264/AVC encoder
+  withX265 ? true, # H.265/HEVC encoder
+  withVmaf ? false, # Perceptual video quality assessment algorithm
+}:
+# av1an requires at least one encoder
+assert lib.assertMsg
+  (lib.elem true [
+    withAom
+    withSvtav1
+    withRav1e
+    withVpx
+    withX264
+    withX265
+  ])
+  "At least one encoder is required!";
+symlinkJoin {
```

Comment:
is this necessary ? why not just wrap in "av1an-unwrapped"

Reply:
i wouldn't call it necessary, moreso a convivence. using a wrapper here allows for configuring what encoders are(n't) used without requiring a - much more expensive - rebuild of `av1an` itself. this puts less burden on end users, as well as hydra (in the event of encoder updates for example)

Reply:
thank you for clarification